CMakeLists.txt

```diff
@@ -0,0 +1,54 @@
+# Specify the minimum version.
+cmake_minimum_required ( VERSION 3.0 )
+
+# Specify the project info.
+project ( trixi VERSION 0.1.0 DESCRIPTION "Interface library for using Trixi.jl from C/C++/Fortran" )
+
+# Enable fortran
+enable_language(Fortran)
+
+
+
+# Library target
+add_library ( ${PROJECT_NAME} SHARED
+    src/trixi.c
+    src/trixi.h
+)
+
+# Include directories, private
+target_include_directories ( ${PROJECT_NAME} PRIVATE src )
+
+# Version info
+set_target_properties ( ${PROJECT_NAME} PROPERTIES VERSION ${PROJECT_VERSION} )
+
+# Version info for the shared object
+set_target_properties ( ${PROJECT_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} )
+
+# Include directories, public for actual users
+set_target_properties ( ${PROJECT_NAME} PROPERTIES PUBLIC_HEADER src/trixi.h )
+
+
+
+# Additional cmake modules
+list ( APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake/" )
+
+# Find Julia
+find_package( Julia REQUIRED )
+
+include_directories( ${JULIA_INCLUDE_DIRS} )
+
+
+
+# Find MPI
+find_package( MPI REQUIRED )
+
+
+
+# Build examples
+add_executable(fortran_hello_world src/examples/fortran_hello_world.f90)
```

Comment:
I recommend to not put the examples folder inside the src dir but as a new toplevel directory in the repo. Then, the examples folder should get its on CMakeLists.txt and this stuff here should be removed. That way, you have complete isolation of the library from the examples. This would be a great help for others who want to use the Trixi.jl library, since they now can see what they need to do to link their own program to our library. That is,they can use your example as a blueprint.